src/lib/components/MainPanel.js

```diff
@@ -66,18 +61,12 @@ export default class MainPanel extends React.Component {
     scheduleSnoozedTab(time);
   }
 
-  handleTimeSelect(value) {
-    this.setState({ dateChoice: value });
-  }
-
   closeTimeSelect() {
     this.setState({ datepickerActive: false });
   }
 
-  confirmTimeSelect() {
-    const { dateChoice } = this.state;
+  confirmTimeSelect(dateChoice) {
     const { scheduleSnoozedTab } = this.props;
-
     if (!dateChoice) { return; }
```

Comment:
Okay, one more update for this - I added a disabled state & style for the snooze button that triggers when the selected time is in the past. Just did a quick & dirty opacity drop, but we can refine that if it's not copacetic